File: src/lib/classes/Chambre.js
/**
 * Classe Chambre - Projet EMSI par Oussama SAJJI
 * 
 * Une chambre est caracterisee par son numero et un numero de telephone
 * L'hotel contient environ 80 chambres
 */

class Chambre {
  // Constructeur par defaut
  constructor(id = null, numero = "", telephone = "") {
    this._id = id;
    this._numero = numero;
    this._telephone = telephone;
  }

  // ═══════════════════════════════════════════════════════════════════════════
  // GETTERS
  // ═══════════════════════════════════════════════════════════════════════════

  get id() {
    return this._id;
  }

  get numero() {
    return this._numero;
  }

  get telephone() {
    return this._telephone;
  }

  // ═══════════════════════════════════════════════════════════════════════════
  // SETTERS
  // ═══════════════════════════════════════════════════════════════════════════

  set id(value) {
    this._id = value;
  }

  set numero(value) {
    this._numero = value;
  }

  set telephone(value) {
    this._telephone = value;
  }

  // ═══════════════════════════════════════════════════════════════════════════
  // METHODES
  // ═══════════════════════════════════════════════════════════════════════════

  /**
   * Methode de saisie - remplit les attributs de la chambre
   * @param {Object} donnees - {numero, telephone}
   */
  saisir(donnees) {
    if (donnees.id !== undefined) this._id = donnees.id;
    if (donnees.numero !== undefined) this._numero = donnees.numero;
    if (donnees.telephone !== undefined) this._telephone = donnees.telephone;
  }

  /**
   * Methode d'affichage - retourne une representation textuelle de la chambre
   * @returns {string}
   */
  afficher() {
    return `
┌─────────────────────────────────────────────┐
│ CHAMBRE #${this._id || "N/A"}
├─────────────────────────────────────────────┤
│ Numero    : ${this._numero}
│ Telephone : ${this._telephone}
└─────────────────────────────────────────────┘`;
  }

  /**
   * Compare deux chambres selon leur numero (code)
   * Retourne True si meme numero, False sinon
   * @param {Chambre} autreChambre
   * @returns {boolean}
   */
  comparer(autreChambre) {
    if (!(autreChambre instanceof Chambre)) return false;
    return this._numero === autreChambre.numero;
  }

  /**
   * Compare par ID
   * @param {Chambre} autreChambre
   * @returns {boolean}
   */
  comparerParId(autreChambre) {
    if (!(autreChambre instanceof Chambre)) return false;
    return this._id === autreChambre.id;
  }

  /**
   * Convertit en objet simple (pour Prisma/JSON)
   * @returns {Object}
   */
  toObject() {
    return {
      id: this._id,
      numero: this._numero,
      telephone: this._telephone,
    };
  }

  /**
   * Cree une instance Chambre depuis un objet (ex: resultat Prisma)
   * @param {Object} obj
   * @returns {Chambre}
   */
  static fromObject(obj) {
    return new Chambre(obj.id, obj.numero, obj.telephone);
  }
}

module.exports = Chambre;

File: java/src/hotel/Chambre.java
package hotel;

import java.util.Scanner;

/**
 * ╔═══════════════════════════════════════════════════════════════════════════╗
 * ║                         CLASSE CHAMBRE                                     ║
 * ╠═══════════════════════════════════════════════════════════════════════════╣
 * ║  Représente une chambre de l'hôtel (80 chambres max)                      ║
 * ║  Projet EMSI - Préparé par Oussama SAJJI                                  ║
 * ╚═══════════════════════════════════════════════════════════════════════════╝
 */
public class Chambre {
    
    // ═══════════════════════════════════════════════════════════════════════════
    // ATTRIBUTS
    // ═══════════════════════════════════════════════════════════════════════════
    
    private int id;
    private String numero;      // Ex: "101", "205", "802"
    private String telephone;   // Téléphone de la chambre
    
    // ═══════════════════════════════════════════════════════════════════════════
    // CONSTRUCTEURS
    // ═══════════════════════════════════════════════════════════════════════════
    
    /**
     * Constructeur par défaut
     */
    public Chambre() {
        this.id = 0;
        this.numero = "";
        this.telephone = "";
    }
    
    /**
     * Constructeur avec paramètres
     * @param id Identifiant unique
     * @param numero Numéro de la chambre (ex: "101")
     * @param telephone Numéro de téléphone de la chambre
     */
    public Chambre(int id, String numero, String telephone) {
        this.id = id;
        this.numero = numero;
        this.telephone = telephone;
    }
    
    // ═══════════════════════════════════════════════════════════════════════════
    // GETTERS
    // ═══════════════════════════════════════════════════════════════════════════
    
    public int getId() {
        return id;
    }
    
    public String getNumero() {
        return numero;
    }
    
    public String getTelephone() {
        return telephone;
    }
    
    // ═══════════════════════════════════════════════════════════════════════════
    // SETTERS
    // ═══════════════════════════════════════════════════════════════════════════
    
    public void setId(int id) {
        this.id = id;
    }
    
    public void setNumero(String numero) {
        this.numero = numero;
    }
    
    public void setTelephone(String telephone) {
        this.telephone = telephone;
    }
    
    // ═══════════════════════════════════════════════════════════════════════════
    // METHODES
    // ═══════════════════════════════════════════════════════════════════════════
    
    /**
     * Saisie interactive d'une chambre
     * @param scanner Scanner pour la lecture
     */
    public void saisir(Scanner scanner) {
        System.out.println("\n--- Saisie Chambre ---");
        
        System.out.print("Numéro de chambre: ");
        this.numero = scanner.nextLine();
        
        System.out.print("Téléphone: ");
        this.telephone = scanner.nextLine();
    }
    
    /**
     * Affichage des informations de la chambre
     */
    public void afficher() {
        System.out.println("╔════════════════════════════════╗");
        System.out.println("║  Chambre " + String.format("%-20s", numero) + " ║");
        System.out.println("╠════════════════════════════════╣");
        System.out.println("║  ID: " + String.format("%-24d", id) + " ║");
        System.out.println("║  Tél: " + String.format("%-23s", telephone) + " ║");
        System.out.println("╚════════════════════════════════╝");
    }
    
    /**
     * Vérifie si deux chambres sont identiques (même numéro)
     * @param autre Chambre à comparer
     * @return true si même numéro
     */
    public boolean equals(Chambre autre) {
        if (autre == null) return false;
        return this.numero.equals(autre.numero);
    }
    
    @Override
    public String toString() {
        return "Chambre " + numero + " (Tél: " + telephone + ")";
    }
}
